### ISSUE TITLE
Go to definition goes all the way up in define chain

### ISSUE THREAD

**Issue by MaJerle** [microsoft/vscode-cpptools#8713]
Bug type: Language Service

**Describe the bug**
- OS and Version: Windows 10.0.19042
- VS Code Version: 1.63.2
- C/C++ Extension Version: 1.8.0
- Other extensions you installed (and if the issue persists after disabling them):
- If using SSH remote, specify OS of remote machine:
- A clear and concise description of what the bug is, including information about the workspace (i.e. is the workspace a single project or multiple projects, size of the project, etc).

Intellisense and browser doesn't properly go to definition. If your define is constructed by different defines, it goes all up to the very first one, which is not expected behavior at all.

**Steps to reproduce**

Consider this code
```c
#define DEF_1               0x12345678
#define DEF_2               DEF_1
#define DEF_3               DEF_2

void
my_func(void) {
    /* Doesn't really matter */
    DEF_3;
}
```

When code is compiled, it works, but does nothing, super useless.
Now doing `CTRL + left-click` on DEF_3 in `my_func`, I'd suppose to go to `DEF_3` definition in the source, instead it jumps to `DEF_1` instead.

In this particular case it may be easy to handle, but in very complex applications, it results to super hard to find the link of definition. Doesn't work at all and unable to quickly access define to modify it.

**Expected behavior**

Go to actual define definition, not to its predecessors.

**Code sample and logs**

- Code sample

- Configurations in `c_cpp_properties.json`
```
{
    "version": 4,
    "configurations": [
        {
            "name": "STM32H7xx ARM none eabi GCC",
            "includePath": [],
            "defines": [],
            "compilerPath": "c:\\msys64\\mingw64\\bin\\gcc.exe",
            "cStandard": "gnu17",
            "cppStandard": "gnu++14",
            "intelliSenseMode": "windows-gcc-x86",
            "configurationProvider": "ms-vscode.cmake-tools"
        }
    ]
}
```
- Logs from running `C/C++: Log Diagnostics` from the VS Code command palette

- Logs from [the language server logging](https://code.visualstudio.com/docs/cpp/enable-logging-cpp#_enable-logging-for-the-language-server)

**Screenshots**

Before click, CTRL + mouse hover produces this:
![pic1](https://user-images.githubusercontent.com/405717/150852447-db2733a7-1426-4119-8cde-e019c3c868b4.png)

After click
![pic2](https://user-images.githubusercontent.com/405717/150852468-95e4f59a-4ae4-46aa-afee-9aa5c7320d88.png)

**Additional context**

**Comment by Colengms** [comment]
Hi @MaJerle .  I think this may be a duplicate of https://github.com/microsoft/vscode-cpptools/issues/3579 .  Can you confirm?

**Comment by sean-mcmanus** [comment]
@Colengms I don't think it's related to #3579 -- it may be  duplicate of https://github.com/microsoft/vscode-cpptools/issues/8701 though (I can confirm this later in my branch that has the fix).

**Comment by MaJerle** [comment]
Thanks to come back.

- It is not linked to #3579 -> this one is about displayed value when you hover
- Seems more duplicate of #8701 

Essentially, `CTRL + click` should jump to actual definition, it should not automatically resolve recursively other definitions.
Currently you cannot find definition of your macro easily and if you are writing the code and would like to correct macro, its a nightmare.

Does this make sense?

**Comment by MaJerle** [comment]
See video

https://user-images.githubusercontent.com/405717/150859645-25045291-32fd-4a35-80d2-86a0875606d7.mp4

**Comment by sean-mcmanus** [comment]
@Colengms It's not fixed with the current batch of IntelliSense fixes and the bug repros in VS too -- only with the standards compliant preprocessor /Zc:preprocessor. This the screenshot when hovering over the front of the DEF3 usage after doing a Go to Def:
![image](https://user-images.githubusercontent.com/19859882/150863551-a3ad8feb-8417-4cd2-8790-04033aa8739c.png)

**Comment by sean-mcmanus** [comment]
Actually, it does appear to be a duplicate, but there's some problem with the "fix" we're investigating still...

**Comment by sean-mcmanus** [comment]
We'll track this as a duplicate of https://github.com/microsoft/vscode-cpptools/issues/8577 (not the other issue mentioned) and confirm this repro is also fixed when that fix gets in (we incorrectly thought that issue was fixed).

**Comment by sean-mcmanus** [comment]
FYI, this is fixed with 1.10.3 (pre-release).
